File: modules/layout/source/types/layout_types/styles/IStyle.ts
// Интерфес стиля
export default interface IStyle {
    id: string // id стиля
    type: StyleType // Название стиля
    value: any // Значение
    getName(): string // Получение имени стиля
    getId(): string // Получение id стиля
    getValue(): any // Получение значения стиля в виде интерпретируемого типа
    setValue(value: any): void // Установка значения
    getString(): string // Получение строки
    getForm(): any // Получение формы для редактирования
}

export enum StyleType {
    WIDTH = 'width',
    HEIGHT = 'height',
}

File: modules/layout/app/types/layout_types/styles/IStyle.go
package styles

type BaseStyle struct {
	Id   string
	Type StyleType
}

// Общий интерфейс для стилей
type IStyle interface {
	GetName() StyleType
	GetValue() interface{}
}

// тип стиля
type StyleType string

const (
	WIDTH  StyleType = "width"
	HEIGHT StyleType = "height"
	COLOR  StyleType = "color"
)
